R/vec.R

```diff
@@ -0,0 +1,138 @@
+#' Concatenate objects in a single vector
+#'
+#' @description
+#'
+#' `vec()` is like [base::c()] but uses an extensible mechanism for
+#' coercions of elements to concatenate.
+#'
+#' Note that only the binary case is guaranteed to be symmetric. The
```

Comment:
I still don't understand why this is true

Reply:
For instance with these methods:

```
quux -> foo
foo -> bar
bar -> quux
```

We get the following coercion paths:

```
vec(quux, foo, bar)
      |-> foo
           |-> bar

vec(foo, bar, quux)
     |-> bar
          |-> quux
```